provider/manifest/service.go

```diff
@@ -269,11 +266,17 @@ func fetchExistingLeases(_ context.Context, session session.Session) ([]event.Le
 
 	items := make([]event.LeaseWon, 0, len(leases))
 	for _, lease := range leases {
-		dgroup, err := session.Client().Query().Group(lease.GroupID())
+		res, err := session.Client().Query().Group(
+			context.Background(),
+			&dtypes.QueryGroupRequest{
+				ID: lease.LeaseID.BidID().GroupID(),
```

Comment:
We should have `DeploymentID()`, `GroupID()`, `OrderID()` methods on `LeaseID` too.